src/OpenTelemetry.AutoInstrumentation/Configuration/Settings.cs

```diff
@@ -24,56 +24,42 @@ namespace OpenTelemetry.AutoInstrumentation.Configuration;
 /// </summary>
 internal abstract class Settings
 {
-    /// <summary>
-    /// Initializes a new instance of the <see cref="Settings"/> class
-    /// using the specified <see cref="IConfigurationSource"/> to initialize values.
-    /// </summary>
-    /// <param name="source">The <see cref="IConfigurationSource"/> to use when retrieving configuration values.</param>
-    protected Settings(IConfigurationSource source)
-    {
-        if (source == null)
-        {
-            throw new ArgumentNullException(nameof(source));
-        }
-
-        OtlpExportProtocol = GetExporterOtlpProtocol(source);
-    }
-
     /// <summary>
     /// Gets the the OTLP transport protocol. Supported values: Grpc and HttpProtobuf.
     /// </summary>
-    public OtlpExportProtocol? OtlpExportProtocol { get; }
+    public OtlpExportProtocol? OtlpExportProtocol { get; private set; }
 
     public static T FromDefaultSources<T>()
-        where T : Settings
+        where T : Settings, new()
```

Comment:
Could be further optimized, so there will be a single configuration instance per setting object. Should be noted, it will be still short living object and there is a conflict with lazy pattern.

Additional + would be to take account to make a possible extensibility point, so plugins can define proprietary configuration source.

Reply:
I would keep minimal extension points. As long as there is no request for that, we should avoid it.

Reply:
Sure, the ideas was just not to rewrite it like it will be a dead end, but to have some open mind when doing it.